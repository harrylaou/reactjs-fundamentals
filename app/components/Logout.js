import React from 'react'
import FlatButton from 'material-ui/FlatButton'

const Logout = ({ logoutUser, color }) => (
  <FlatButton style={{color}} label="Logout" onClick={()=>{ logoutUser() }}/>
)


Logout.propTypes = {
  color: React.PropTypes.string.isRequired,
  logoutUser: React.PropTypes.func.isRequired,
}

export default Logout
