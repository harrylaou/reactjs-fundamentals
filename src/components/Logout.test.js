import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Logout from './Logout'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

describe('<Logout />', () => {
  it('should render with default props', () => {
    const logoutUser = jest.fn();
    const props = {
      color: '',
      logoutUser
    }

    const wrapper = shallow(
      <Logout {...props} />,
      {context: {
        muiTheme: getMuiTheme()
      }}
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.find('FlatButton').simulate('click')

    expect(logoutUser).toBeCalled()
  })
})