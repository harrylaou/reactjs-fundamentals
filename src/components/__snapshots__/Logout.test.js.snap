// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render with default props 1`] = `
<FlatButton
  disabled={false}
  label="Logout"
  labelPosition="after"
  labelStyle={Object {}}
  onClick={[Function]}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  primary={false}
  secondary={false}
  style={
    Object {
      "color": "",
    }
  }
/>
`;
